pageserver/src/tenant/timeline.rs

```diff
@@ -1860,6 +1867,31 @@ impl Timeline {
         Ok(lease)
     }
 
+    #[instrument(skip(_ctx), ret(Debug), err(Debug))]
+    pub(crate) fn lease_standby_horizon(
+        &self,
+        lease_id: String,
+        lsn: Lsn,
+        _ctx: &RequestContext,
+    ) -> anyhow::Result<SystemTime> {
+        if self
+            .feature_resolver
+            .evaluate_boolean("standby-horizon-leases-track-disable")
+            .is_ok()
+        {
+            // Use feature-flagging as a fail-safe in case something is wrong with the data structure (memory consumption, etc.)
+            return Ok(SystemTime::now()
+                .checked_add(Duration::from_secs(5 * 60))
+                .unwrap());
+        }
+        let applied_gc_cutoff_lsn =
+            todo!("think about boundary conditions? we didn't have any before though");
+        let length = todo!("duplicate init lease deadline logic?");
```

Comment:
This TODO is about a) need for tenant_config fields analogous to lsn leases, and b)  the need to implement the init lease deadline after startup/attach, same as lsn leases.

It is absolutely necessary for correctness because, like lsn leases, standby_horizon leases are not persistent.

The existing standby_horizon propagation mechanism is also not persistent and in theory has the same problem that would need to be fixed by initial lease deadline. But in practice, the propagation via sk push_loop -> broker is (probably, haven't checked) more timely than the lease renewal period I have in mind.